sdk/storage/src/datalake/service_client.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "datalake/service_client.hpp"
+
+#include "common/common_headers_request_policy.hpp"
+#include "common/constant.hpp"
+#include "common/shared_key_policy.hpp"
+#include "common/storage_common.hpp"
+#include "datalake/file_system_client.hpp"
+#include "http/curl/curl.hpp"
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  ServiceClient ServiceClient::CreateFromConnectionString(
+      const std::string& connectionString,
+      const ServiceClientOptions& options)
+  {
+    auto parsedConnectionString = ParseConnectionString(connectionString);
+
+    std::string accountName;
+    std::string accountKey;
+    std::string blobEndpoint;
+    std::string datalakeEndpoint;
+    std::string EndpointSuffix;
+    std::string defaultEndpointsProtocol = Details::c_PathDnsSuffixDefault;
```

Comment:
if file_system_client, path_client and service_client all three have these same fields, and they all can parse from a connectionString, I would suggest to create an abstract class on top of them and move those fields to there.
Then create a function to parse and load those fields from connection strings and re-use it the three clasees..
Unless there's something I missed.

Reply:
Yeah, that's what i'm thinking about. But i don't have too much time now and this is not a priority. I will do this in the future.

Reply:
Created an issue to track and resolving this comment: https://github.com/Azure/azure-sdk-for-cpp/issues/192